### PR TITLE
buildkite-agent-metrics: init at 5.1.0

### PR DESCRIPTION
**Pull request by craigem**
###### Motivation for this change

A command-line tool for collecting Buildkite agent metrics, focusing on enabling auto-scaling.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

None yet.

### CODE REVIEW

File: pkgs/servers/monitoring/buildkite-agent-metrics/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{ lib, go, buildGoPackage, fetchFromGitHub }:
+
+let
+  goPackagePath = "github.com/buildkite/buildkite-agent-metrics";
```
Comment (alexarice) [inline pkgs/servers/monitoring/buildkite-agent-metrics/default.nix]: Can't see a reason not to just put this in the derivation

File: pkgs/servers/monitoring/buildkite-agent-metrics/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+       -X ${t}.GoVersion=${lib.getVersion go}
+  '';
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "A command-line tool (and Lambda) for collecting Buildkite agent metrics";
+    homepage = "https://github.com/buildkite/buildkite-agent-metrics";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (alexarice) [inline pkgs/servers/monitoring/buildkite-agent-metrics/default.nix]: Are you not willing to be a maintainer for this software? there is a push to try and reduce the number of unmaintained packages in nixpkgs

Comment (craigem) [reply pkgs/servers/monitoring/buildkite-agent-metrics/default.nix]: Hi @alexarice, thanks for the reviews.

I am happy to be maintainer but ofBorg failed when I put my name in it, so a second commit removed my name so I could see the tests pass.

File: pkgs/servers/monitoring/buildkite-agent-metrics/default.nix (hunk 1 above)
Comment (craigem) [reply pkgs/servers/monitoring/buildkite-agent-metrics/default.nix]: Great point @alexarice - I was just rather blindly following the convention of related Go packages, such as Prometheus. I've corrected this in the most recent commit.

File: pkgs/servers/monitoring/buildkite-agent-metrics/default.nix (hunk 2 above)
Comment (alexarice) [reply pkgs/servers/monitoring/buildkite-agent-metrics/default.nix]: Have you got an entry in `maintainers.nix`?

**Comment by doronbehar** [comment]
@craigem your PR should have eventually two commits:

```
maintainers: add <me>
<package-name>: init at <version>
```

As for the CI error, just make sure you don't put a typo in that `maintainers-list.nix` file and you will pass the CI, it's not that hard.

**Comment by SuperSandro2000** [comment]
Going to close this PR because @craigem did not react in the last months.

File: pkgs/servers/monitoring/buildkite-agent-metrics/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  buildFlagsArray = let
+    t = "${goPackagePath}/vendor/github.com/buildkite/buildkite-agent-metrics/version";
+  in ''
+    -ldflags=
+       -X ${t}.Version=${version}
+       -X ${t}.Revision=unknown
+       -X ${t}.Branch=unknown
+       -X ${t}.BuildUser=nix@nixpkgs
+       -X ${t}.BuildDate=unknown
+       -X ${t}.GoVersion=${lib.getVersion go}
```
Comment (grahamc) [inline pkgs/servers/monitoring/buildkite-agent-metrics/default.nix:29]: Does a multi-line string actually work for this field?

**Pull request closed without merging**